rivgraph/ln_utils.py

```diff
@@ -868,11 +870,16 @@ def remove_all_spurs(links, nodes, dontremove=[]):
                 links, nodes = delete_link(links, nodes, looplink)
                 ct = ct + 1
 
-
+        
         # Remove all the nodes with only two links attached
         links, nodes = remove_two_link_nodes(links, nodes, dontremove)
-
-        if ct == 0:
+        
+         # For single channel (no loops & single outlet) networks,
+        # prevent overpruning beyond a single network with 2 nodes.
+        if (links['n_networks'] == 1) & (len(nodes['conn']) == 2):
+            ct = 0
+        
+        if ct <= ctmax:

```

Comment:
Should this be greater than instead of less than? Seems like stopflag will be set to 1 after the first iteration, halting the pruning prematurely.

Reply:
If you run this once, say getting ct = 55, and ctmax = 0, it shouldn't fail. Can discuss more on this as it helps address [Issue 47](https://github.com/jonschwenk/RivGraph/issues/47)